### PR TITLE
Organize output and consider re-use of intermediate artifacts in `pairwise_evaluation_ants`

### PR DESCRIPTION
**Pull request by jamesfishbaugh**

### CODE REVIEW

**Comment by jamesfishbaugh** [comment]
This PR also addresses issue #18 to process only a specified subset of tracts.

**Comment by jamesfishbaugh** [comment]
The readme should be up to date with the changes to the usage. Let me know if there is anything ambiguous that can be improved.

File: evaluation_framework/pairwise_evaluation_ants.py [hunk 1]
```diff
@@ -17,38 +21,33 @@
...
 source_fiber_path = Path(os.path.abspath(args.source_fiber_dir))
 
 target_path = Path(os.path.abspath(args.target))
 target_without_ext = str(target_path)[:str(target_path).rfind(''.join(target_path.suffixes))]
-target_bval_path = Path('%s.bval' %(target_without_ext))
-target_bvec_path = Path('%s.bvec' %(target_without_ext))
-target_mask_path = Path(os.path.abspath(args.target_mask))
 target_fiber_path = Path(os.path.abspath(args.target_fiber_dir))
 
 output_path = Path(os.path.abspath(args.output_dir))
```
Comment (ebrahimebrahim) [inline evaluation_framework/pairwise_evaluation_ants.py:53]: Need to replace `output_dir` by `output_base_dir` here

**Review by ebrahimebrahim — APPROVED** [review]
I tried this out on a couple of images from our evaluation sample. Here is the command that I ran:

```sh
datadir="/data/ebrahim-data/abcd/registration-experiments/evaluation-2024-05"
source_subject="sub-NDARINV7RH75L3L_ses-2YearFollowUpYArm1_run-01_dwi"
target_subject="sub-NDARINV01Z8HAPV_ses-2YearFollowUpYArm1_run-01_dwi"
python pairwise_evaluation_ants.py $datadir/dti_output/$source_subject/${source_subject}_fa.nii.gz $datadir/tractseg_output/$source_subject/TOM_trackings/ $datadir/dti_output/$target_subject/${target_subject}_fa.nii.gz $datadir/tractseg_output/$target_subject/TOM_trackings/  ~/Desktop/testing_issue_16/ example_experiment_name2
```

It worked nicely! I like the idea of an experiment name and data json. I am not sure though about how it will work when we go to run evaluations of many metrics over many registration methods -- we might end up using the api you have made for that. I think the experiment tracking will be helpful when we are testing individual steps mainly.

Approving -- i will just fix the one typo before merging

**Pull request merged**